encoding/ssz/merkleize.go

```diff
@@ -196,3 +197,42 @@ func ConstructProof(hasher Hasher, count, limit uint64, leaf func(i uint64) []by
 
 	return
 }
+
+// MerkelizeVector uses our optimized routine to hash a list of 32-byte
+// elements.
+func MerkelizeVector(elements [][32]byte, length uint64) [32]byte {
+	depth := Depth(length)
+	// Return zerohash at depth
+	if len(elements) == 0 {
+		zerohash := trie.ZeroHashes[depth]
```

Comment:
Shouldn't the case `length = 1` be treated separately? In this case I thought a vector of length one would be hashed as just the one chunk itself. So if `len(elements) = 0` we should return `ZeroHashes[0]` but this would return `ZeroHashes[1]`.